### PR TITLE
Fix CI error due to autoformat

### PR DESCRIPTION
**Pull request by naftalibeder**
Fixes the CI error in #2167. The problem was that `plugin/build/withMapbox.js` was being generated and then autoformatted, so a "stable" JS file could never actually get committed.

Ideally, I think that whole `build` directory should be ignored from git, but I didn't want to complicate things for now.

### CODE REVIEW

**Comment by mfazekas** [comment]
I think the issue with CI is that you have different typescript compiler locally which transpiles different catch than on CI server. 

The easiest fix would be reverting `catch (_a) {` to `catch {`

https://github.com/rnmapbox/maps/pull/2167/files#diff-8c0da0f7d6c1fd23bf87929c99f86a6daeb4a2cc1beb4835d1fa7ee56b24b1e7L17

> Ideally, I think that whole build directory should be ignored from git, but I didn't want to complicate things for now.

Not sure if expo plugins needs to be transpired, or not.

**Comment by naftalibeder** [comment]
So I don't actually have typescript installed globally - my scripts are using the version in this project's `package.json` (which admittedly I updated, but shouldn't the CI mirror that?).

It seems like the CI is using an older version than now specified. Is there a place we can change that?

File: plugin/src/withMapbox.ts [hunk 1]
```diff
@@ -22,8 +22,8 @@ let pkg: { name: string; version?: string } = {
 };
 try {
   pkg = require('@rnmapbox/maps/package.json');
-} catch {
-  // empty catch block
+} catch (e) {
+  console.warn('Could not find an existing package.json file');
```
Comment (mfazekas) [inline plugin/src/withMapbox.ts:26]: I'm not familiar enough with expo plugins, so I'm not sure if it's better to give a warning, or not.

File: example/package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "purge:ios": "rm -rf ios/Pods/* ios/build ~/Library/Caches/CocoaPods ~/Library/Developer/Xcode/DerivedData && pod cache clean --all",
     "purge:js": "rm -rf node_modules && yarn cache clean && watchman watch-del-all",
     "purge": "yarn purge:js && yarn purge:android && yarn purge:ios",
-    "type:check": "tsc --noEmit"
+    "type:check": "npx tsc --noEmit"
```
Comment (mfazekas) [inline example/package.json:15]: 👍

**Review by mfazekas — CHANGES_REQUESTED** [review]
I'd remove the console.warn otherwise looks good.

**Comment by naftalibeder** [comment]
> I'd remove the console.warn

Having a statement inside the `catch` block prevents the compiler from (unpredictably) creating/renaming the error variable. Would you be okay with something like

```
try {
  pkg = require('@rnmapbox/maps/package.json');
} catch (e) {
  console.log('No existing package.json file found, using default configuration');
}
```

**Comment by mfazekas** [comment]
> > I'd remove the console.warn
> 
> Having a statement inside the `catch` block prevents the compiler from (unpredictably) creating/renaming the error variable. Would you be okay with something like
> 
> ```
> try {
>   pkg = require('@rnmapbox/maps/package.json');
> } catch (e) {
>   console.log('No existing package.json file found, using default configuration');
> }
> ```

Yeah, sorry I still don't like the idea of altering the behaviour because of a software infrastructure issue. I thought it's just a tsc version difference issue, but it doesn't seems so at a second glance.

Let's just leave that generated file as our CI likes.

https://github.com/rnmapbox/maps/commit/4a5809d45833f3b2844aef83a3632f4d628a081e

**Comment by naftalibeder** [comment]
Ok seems good.

It does seem like a shame to downgrade something when the upgrade is actually encouraging better usability (not great to silently swallow an error), but I understand your feeling about the reasoning.

Good to merge?

**Comment by mfazekas** [comment]
> Ok seems good.
> 
> It does seem like a shame to downgrade something when the upgrade is actually encouraging better usability (not great to silently swallow an error), but I understand your feeling about the reasoning.
> 
> Good to merge?

Sorry I'm a bit confused. 

So my understanding that there is some issue that causes  typescript to compile different code on CI than on locally, and that can cause CI failure if you commit your local version of the plugin's `build` folder. That is less than ideal, but for now please ignore, and just don't commit to plugin's `build` folder, unless you're working on that. (For me this doesn't seems to be an issue as on my local copy `withMapbox.js` has later date than `withMatbox.ts` so it's not rebuilding)

I think the proper solution for this would be to remove `build` folder from plugin if possible - but not sure about that. Another one is commiting a `yarn.lock` file if that's fixes makes the tsc compile the same on CI than locally.

**Comment by naftalibeder** [comment]
Sorry for the confusion.

> there is some issue that causes typescript to compile different code on CI than on locally, and that can cause CI failure if you commit your local version of the plugin's build folder.

This is correct. But that was because I upgraded the project's typescript version, which didn't seem to propagate to the CI.

But the simplest way to fix this that doesn't involve me (and presumably other people) avoiding committing one file, in my opinion, is what I did, which was to downgrade to the previously used version of typescript. That way, doing a build locally matches what the CI does.

This effectively reverts the problem commit. If this isn't okay for you, can you explain further why?

**Comment by mfazekas** [comment]
> Sorry for the confusion.
> 
> > there is some issue that causes typescript to compile different code on CI than on locally, and that can cause CI failure if you commit your local version of the plugin's build folder.
> 
> This is correct. But that was because I upgraded the project's typescript version, which didn't seem to propagate to the CI.

I think the CI picked up the latest ts version just fine. 

I've made a branch where I've added a dump on version and it was 4.8.2:
https://github.com/rnmapbox/maps/pull/2195

https://github.com/rnmapbox/maps/actions/runs/3009586590/jobs/4834748131

```sh
$ tsc --version && tsc --build plugin
[14](https://github.com/rnmapbox/maps/actions/runs/3009586590/jobs/4834748131#step:11:15)
Version 4.8.2
```

> 
> But the simplest way to fix this that doesn't involve me (and presumably other people) avoiding committing one file, in my opinion, is what I did, which was to downgrade to the previously used version of typescript. That way, doing a build locally matches what the CI does.

Again I'm not convinced that the issue was caused by different ts versions. 

> 
> This effectively reverts the problem commit. If this isn't okay for you, can you explain further why?

We can downgrade the ts version, but i don't see any evidence that it's related to the issue.

**Comment by naftalibeder** [comment]
@mfazekas Bump?

**Comment by mfazekas** [comment]
> @mfazekas Bump?
Yep it’s strange that it sometimes generates different catch statement.

Pls revert withMapbox.ts, .js changes, otherwise good to go!

**Comment by naftalibeder** [comment]
Ok... reverted!

**Pull request merged**